### PR TITLE
Add dataprovider helpers to CoreServices

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview

This change makes it easier to access data providers registered to core services. In particular it adds explicit helpers for the input system and spatial awareness system.

This is done by adding helper methods to the CoreServices.cs class
`GetInputSystemDataProvider<T>()`
`GetSpatialAwarenessSystemDataProvider<T>()`
`GetDataProvider<T>(service)`

This change also updates use of this pattern across the MRTK code base as well as updates the documentation to reflect this new, easier to use code path.

## Changes
- Fixes: #7134

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/CoreServices.cs [hunk 1]
```diff
@@ -102,6 +102,41 @@ public static bool ResetCacheReference(Type serviceType)
...
+        /// </summary>
+        public static T GetSpatialAwarenessSystemDataProvider<T>() where T : IMixedRealitySpatialAwarenessObserver
+        {
+            return GetDataProvider<T>(SpatialAwarenessSystem);
+        }
+
+        /// <summary>
+        /// Gets first matching data provider of provided type T registered to the provided mixed reality service.
+        /// </summary>
+        /// <typeparam name="T">Type of data provider to return. Must implement and/or extend from <see cref="IMixedRealityDataProvider" /> </typeparam>
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/CoreServices.cs]:
```suggestion
        /// <typeparam name="T">Type of data provider to return. Must implement and/or extend from <see cref="IMixedRealityDataProvider" /></typeparam>
```

**Comment by wiwei** [comment]
lgtm, modulo a few doccomments that hare non-resolving crefs i.e. 

/// Gets first matching <see cref="IMixedRealityInputDeviceManager"/> or extension thereof for CoreServices.InputSystem

and 

        /// Gets first matching <see cref="IMixedRealitySpatialAwarenessObserver"/> or extension thereof for CoreServices.SpatialAwarenessSystem

Also just to make sure I understand, this also adds a safeguard in the SA inspector (i.e. in case spatial isn't in existence - I see a couple of new null checks that aren't necessarily directly related to these convenience helpers)

**Comment by Troy-Ferrell** [comment]
> Also just to make sure I understand, this also adds a safeguard in the SA inspector (i.e. in case spatial isn't in existence - I see a couple of new null checks that aren't necessarily directly related to these convenience helpers)

@wiwei , yes I added null checks because technically the spatial awareness system has to implement IMixedRealitySpatialAwarenessSystem but not necessarily IMixedRealityDataProviderAccess. Thus, if someone created their own custom class implementation in this manner, the inspector would throw null exceptions

**Pull request merged**